src/routabled.py

```diff
@@ -0,0 +1,67 @@
+import os
+from pathlib import Path
+from typing import Optional, TypedDict
+
+DEFAULT_TIMEOUT = 300  # 5 minutes
```

Comment:
how long to wait running the config-changed hook before we decide it's not going to finish running the hook.  5m seems like plenty of time to run a charmed hook right?